designs/discontinue_usage_of_kube_rbac_proxy.md

```diff
@@ -0,0 +1,281 @@
+[Plugins API provided by Kubebuilder](./../docs/book/src/plugins/plugins.md),
+to create an [external plugin](./../docs/book/src/plugins/creating-plugins.md)
+to properly integrate the solution with Kubebuilder and provide a helper to allow users to opt-in as please them.
+
+## Open Questions
+
+1. [Network Policies][k8s-doc-networkpolicies] is implemented by the cluster’s CNI. Are we confident that the proposed policy is supported by all the major CNIs in use?
+
+> Besides [Network Policies][k8s-doc-networkpolicies] being part of the core Kubernetes API, their enforcement relies on the CNI plugin installed in
+the Kubernetes cluster. While support and implementation details can vary among CNIs, it seems that the most commonly used ones,
+such as Calico, Cilium, WeaveNet, and Canal, shows to offer  support for NetworkPolicies.
+>
+>Also, there was concern in the past because AWS did not support it. However, this changed,
+>as detailed in their announcement: [Amazon VPC CNI now supports Kubernetes Network Policies](https://aws.amazon.com/blogs/containers/amazon-vpc-cni-now-supports-kubernetes-network-policies/).
+>
+>Moreover, under this proposal users still able to disable/enable this option as please them.
+
+1. NetworkPolicy is a simple firewall and does not provide authn/authz and encryption.
+
+> Yes, that's correct. NetworkPolicy acts as a basic firewall for pods within a Kubernetes cluster, controlling traffic flow at the IP address or port level. However, it doesn't handle authentication (authn), authorization (authz), or encryption directly likw kube-rbac-proxy solution.
+
+1. Could not Kubebuilder maintainers use the shared infrastructure to continue building and promoting those images under the new `registry.k8s.io`?
+
+> We tried to do that, see [here](https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/k8s-staging-kubebuilder.yaml) the recipe implemented.
+> However, it does not work because kube-rbac-proxy is not under the
+> kubernetes umbrella. More over we experiment the GitHub Repository as alternative approach, see the [PR](https://github.com/kubernetes-sigs/kubebuilder/pull/3854) but seems
+> that we are not allowed to use it. Nevertheless, both approaches does not show to sort out all motivations and requirements
+> Ideally, Kubebuilder should not be responsible for maintain and promote third-part artifacts.
+
+1. However, is not Kubebuilder also building and promoting the binaries required to be used within [EnvTest](./../docs/book/src/reference/envtest.md)
+feature implemented in controller-runtime?
+
+> Yes, but it also will need to change. Controller-runtime maintainers are looking for solutions to
+> built those binaries inside its project since it seems part of its domain. This change is likely
+> to be transparent to the community users.
+
+1. Could we not use the Controller-Runtime feature [controller-runtime][cr-pr] which enable secure metrics serving over HTTPS?
```

Comment:
That is a great idea. !!!!

I will add the option of we enhance in a follow up when controller-runtime be updated.
@sbueringer since you are the author of the feature you might want to give a look on that.

If we be able to improve that we can add it in the Kubebuilder in the future for sure
that would be a great option

Reply:
Closing since we updated the proposal within.